libc/semantics/settings.k

```diff
@@ -40,13 +39,15 @@ module C-SETTINGS
      rule cfg:alignofFloat => 4
      rule cfg:alignofDouble => 8
      rule cfg:alignofLongDouble => 16
-     rule cfg:alignofEnum => cfg:alignofInt
      rule cfg:alignofPointer => 4
      // Max of the above.
      rule cfg:alignofMalloc => 16
 
-     rule cfg:intToPointer(_, _) => trap
+     rule cfg:intToPointer(0, _) => NullPointer
+     rule cfg:intToPointer(_, _) => trap [owise]
```

Comment:
technically the C standard does not guarantee the existence of the intptr_t and uintptr_t types, which are the only integer types guaranteed to be large enough to contain a pointer address and represent the same address when casted back. This implementation of C does not contain these types; the one that is in x86_64-linux-gcc-glibc does however. Essentially, settings.k contains implementation-defined behavior, and the behavior when an integer is cast to a pointer and back is implementation-defined.